File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.py
class Solution(object):
    def countPartitions(self, nums):
        t = 0
        for i in range(len(nums)):
            t += nums[i]
        l = 0
        c = 0
        for i in range(len(nums) - 1):
            l += nums[i]
            r = t - l
            if (l % 2 == 0) == (r % 2 == 0):
                c += 1
        return c
            
        

File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.java
class Solution {
    public int countPartitions(int[] nums) {
        int t = 0;
        for (int i = 0;i < nums.length; i++){
            t += nums[i];
        }
        int c = 0;
        int ls = 0;
        for (int i = 0; i < nums.length -1 ; i++){
            ls += nums[i];
            int rs = t - ls;
            if ((ls % 2 == 0) == (rs % 2 == 0)){
                c++;
            }
        }
        return c;
    }
}